packages/api/src/platforms/vtex/clients/search/index.ts

```diff
@@ -199,7 +200,11 @@ export const IntelligentSearch = (
     }
 
     if (hideUnavailableItems !== undefined) {
-      params.append('hideUnavailableItems', hideUnavailableItems.toString())
+      params.append(
+        'hideUnavailableItems',
+        searchHideUnavailableItems?.toString() ??
+          hideUnavailableItems.toString()
+      )
```

Comment:
@lucasfp13 first we check the flag `hideUnavailableItems`, and after we include `searchHideUnavailableItems` param and the flag `hideUnavailableItems` as fallback? should we check the `searchHideUnavailableItems` in the if instead?